src/lib/containers/discussion_forum/ForumThreadEditor.tsx

```diff
@@ -13,58 +15,103 @@ export type ForumThreadEditorProps = {
   initialText?: string
   id: string
   onClose: () => void
+  isReply: boolean
 }
 
 export const ForumThreadEditor: React.FunctionComponent<
   ForumThreadEditorProps
-> = ({ initialText, initialTitle, id, onClose }) => {
+> = ({ initialText, initialTitle, id, onClose, isReply }) => {
   const [title, setTitle] = useState<string>(initialTitle ?? '')
   const [text, setText] = useState<string>(initialText ?? '')
-  const { mutate: updateTitle } = useUpdateThreadTitle()
-  const { mutate: updateMessage } = useUpdateThreadMessage()
-  const { mutate: postThread } = useCreateThread()
+  const { mutate: updateTitle, isLoading: isLoadingTitle } =
+    useUpdateThreadTitle({
+      onSuccess: () => onClose(),
+    })
+  const { mutate: updateMessage, isLoading: isLoadingMessage } =
+    useUpdateThreadMessage({
+      onSuccess: () => onClose(),
+    })
+  const { mutate: postThread, isLoading: isLoadingThread } = useCreateThread({
+    onSuccess: () => onClose(),
+  })
+  const { mutate: postReply, isLoading: isLoadingReply } = usePostReply({
+    onSuccess: () => onClose(),
+  })
```

Comment:
Being super picky but I think this is slightly easier to understand

```suggestion
  const { mutate: createThread, isLoading: isLoadingThread } = useCreateThread({
    onSuccess: () => onClose(),
  })
  const { mutate: createReply, isLoading: isLoadingReply } = usePostReply({
    onSuccess: () => onClose(),
  })
```